ocpi-toolkit-2.1.1-gireve/src/test/kotlin/com/izivia/ocpi/toolkit/samples/common/PlatformCacheRepository.kt

```diff
@@ -7,6 +7,14 @@ import com.izivia.ocpi.toolkit.modules.versions.domain.Version
 open class PlatformCacheRepository: PlatformRepository {
     val platforms = mutableMapOf<String, Platform>()
 
+    override fun savePlatformUrlForTokenA(tokenA: String, platformUrl: String): String? = platforms
+        .toList()
+        .firstOrNull { it.second.tokenA == tokenA }
+        ?.second
+        ?.copy(url = platformUrl)
+        ?.also { platforms[it.url] = it }
+        ?.url
+
```

Comment:
I'm not sure how it works without creating a duplicate: the platforms is a map by url IIUC, but it must be in the map before the call to be found in the list, but at whtat key? And we put it again in the map afterward?

Reply:
It's free to the user to implement `PlatformRepository`. This code is just a sample on how to use the `PlatformRepository`.

Indeed, using a `Map` is not the right choice here, because we cannot guarantee the uniqueness of a `Platform` with its `url` (it's nullable now).

In the example, it initializes a `Platform` with a UUID as URL, that's why it works.

---

I agree, now, this cache sample is not intuitive. Users should refer to the mongo sample as it does not have this issue

Reply:
I tested every credentials module of every OCPI version, and I detected a design issue with `getPlatformByTokenA` and `getPlatformByTokenB`. I updated these methods accordingly and updated the change notes to explain the changes.

It should be good now